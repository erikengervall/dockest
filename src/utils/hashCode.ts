export default (str: string) =>
  Array.from(str).reduce((hash, char) => (Math.imul(31, hash) + char.charCodeAt(0)) | 0, 0)
