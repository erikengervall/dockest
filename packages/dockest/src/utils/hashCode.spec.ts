import { hashCode } from './hashCode'

describe('hashCode', () => {
  describe('happy', () => {
    it('should generate hashCodes deterministically', () => {
      const service = 'postgres1sequelize'
      const service2 = 'postgres1sequelize'

      const result = hashCode(service)
      const result2 = hashCode(service2)

      expect(result).toMatchInlineSnapshot(`-2021595073`)
      expect(result).toEqual(result2)
    })
  })
})
