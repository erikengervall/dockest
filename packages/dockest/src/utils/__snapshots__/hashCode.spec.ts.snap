// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hashCode happy should generate valid hashCode 1`] = `-2021595073`;
